rpmvenv/extensions/python/venv.py

```diff
@@ -108,18 +108,23 @@ def generate(config, spec):
         spec.blocks.files.append('/%{venv_install_dir}')
 
         spec.blocks.install.append('%{venv_cmd} %{venv_dir}')
-        for requirement in config.python_venv.requirements:
-
-            spec.blocks.install.append(
-                '%{{venv_pip}} -r %{{SOURCE0}}/{0}'.format(
-                    requirement,
+        # If we have requirements files, install based on them.
+        if len(tuple(config.python_venv.requirements)) > 0:
+            for requirement in config.python_venv.requirements:
+                spec.blocks.install.append(
+                    '%{{venv_pip}} -r %{{SOURCE0}}/{0}'.format(
+                        requirement,
+                    )
                 )
-            )
+        # Otherwise, use standard Python install.
+        else:
```

Comment:
This actually introduces a change in behaviour that I would consider broken. The assumption made here is that a requirements.txt, should at least one be present, _will_ contain a line to install the target source package as well. This is usually not the case.

I can see the possibility of wanting to build an RPM from only a requirements.txt file, but we would need to find a way of adding it that doesn't impact current users.